agent/api/container.go

```diff
@@ -108,7 +108,8 @@ type Container struct {
 	SentStatusUnsafe ContainerStatus `json:"SentStatus"`
 	sentStatusLock   sync.RWMutex
 
-	KnownExitCode     *int
+	knownExitCode     *int
+	knownExitCodeLock sync.RWMutex
```

Comment:
Can't we have a single `updateLock` for container? I saw there are several locks in this struct, is there a performance concern to have multiple locks for different field? Otherwise it makes the code less complex or running into deadlock issue.